### PR TITLE
Implemented endpoint to add a rss feed to your channel

### PR DESCRIPTION
**Pull request by devos50**
I also did some refactoring of test files (since we have to support other requests than `GET` now).

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis can you review?

**Comment by lfdversluis** [comment]
ok

**Comment by lfdversluis** [comment]
pylint "MyChannelRssFeedsEndpoint.get_my_channel_object: Missing method docstring"

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 1]
```diff
@@ -94,24 +94,51 @@ def render_GET(self, request):
...
-        }, ...]
-    }
+    This class is responsible for handling requests regarding rss feeds in your channel.
     """
 
-    def render_GET(self, request):
+    def get_my_channel_object(self):
         my_channel_id = self.channel_db_handler.getMyChannelId()
         channel_obj = self.session.lm.channel_manager.get_my_channel(my_channel_id)
         if my_channel_id is None or channel_obj is None:
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: Can it happen that my_channel_id is not none but channel_obj is?

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 2]
```diff
@@ -94,24 +94,51 @@ def render_GET(self, request):
...
+        {
+            "rss_feed_url": "http://rssprovider.com/feed.xml"
+        }
+        """
+        channel_obj = self.get_my_channel_object()
+        if channel_obj is None:
+            return MyChannelBaseEndpoint.return_404(request)
+
+        parameters = http.parse_qs(request.content.read(), 1)
+        channel_obj.create_rss_feed(parameters['rss_feed_url'][0])
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: Don't you get a crash if parameters['rss_feed_url'] is empty? What is being returned by the API then? Any error code?

File: Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py [hunk 3]
```diff
@@ -42,6 +35,26 @@ def create_my_channel(self, name, description):
...
+class TestMyChannelEndpoints(AbstractTestMyChannelEndpoints):
+
+    @deferred(timeout=10)
+    def test_my_channel_unknown_endpoint(self):
+        """
+        Testing whether the API returns an error if an unknown endpoint is queried
+        """
+        self.should_check_equality = False
+        return self.do_request('mychannel/thisendpointdoesnotexist123', expected_code=404)
+
```
Comment (lfdversluis) [inline Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py]: What about a test that does a request to 'mychannel/' ? What happens if the param is missing?

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py (hunk 1 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: No that should not happen. I will remove the first check in the conditional.

Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: No need, just return the channel_obj then?

If bla is none:
return none
else:
return bla

is the same as return bla :)

File: Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py (hunk 3 above)
Comment (devos50) [inline Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py]: That's a good idea, I've added such a test.

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py (hunk 2 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: I'm returning error code 500 now together with an error message.

**Comment by devos50** [comment]
@whirm ready for your review

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 4]
```diff
@@ -94,24 +94,56 @@ def render_GET(self, request):
...
+            "rss_feed_url": "http://rssprovider.com/feed.xml"
+        }
+        """
+        channel_obj = self.get_my_channel_object()
+        if channel_obj is None:
+            return MyChannelBaseEndpoint.return_404(request)
+
+        parameters = http.parse_qs(request.content.read(), 1)
+        if 'rss_feed_url' not in parameters or len(parameters['rss_feed_url']) == 0:
+            request.setResponseCode(http.INTERNAL_SERVER_ERROR)
```
Comment (whirm) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: This should be `bad request` (400) not 5XX, there is not an internal server error.

Comment (whirm) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: For future reference: https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

**Comment by whirm** [comment]
Other than that, it looks fine.

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py (hunk 4 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: @whirm thanks!

**Comment by devos50** [comment]
```
======================================================================
ERROR: Uses RSS feed to test metadata.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/test_as_server.py", line 643, in tearDown
    assert boolean, reason
AssertionError: TestMyChannel.test_metadata - Channel did not have torrents - Condition was not satisfied in 60 seconds
```

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm ready for merge

**Pull request merged**